### DISCUSSION TITLE
Add Filter linked to vlan to avoid agent running outside company network

### DISCUSSION THREAD

**Discussion by EC2311** [glpi-project/glpi-agent#777] [Ideas]
### Your idea

Hello

It could be useful that Agent runs only when it is on company networks to avoid discover some private assets (filter by vlan?)

**Comment by g-bougard** [comment]
Hi @EC2311 

to avoid this problem, you can use a DNS name in your glpi-agent `server` configuration which can only be resolved into company networks.

After, you may also have some GLPI server solutions:
 - Add private networks to blacklist and remove still defined private networks so they won't come back
 - not sure, but maybe you can add constraint to import & updates rules against location or network itself

On agent-side, we have no solution to ignore inventory task run from the network context.

**Comment by EC2311** [comment]
Hi @g-bougard 
Thanks For your reply

indeed using a dns (which is done) may be a solution, but when user uses VPN to connect to networkf company, then inventory raises :(
And we don't want to get inventory of private devices which may be connected to computer at home (for instance)

I don't know the content of inventory file when computer is connected to vpn : are all ip raised (local, vpn and so on) : I will check when I will be at home :)
Thanks
Best regards

**Reply by g-bougard** [reply]
You should be able to configure DNS resolution for computers connected via vpn and then don't provide any answer for the server set in agent configuration.
If users should be able to connect anyway to GLPI, you should be able to use a dedicated DNS alias in agent configuration and only not answer for this alias.
